core/src/poh_recorder.rs

```diff
@@ -182,33 +176,28 @@ impl PohRecorder {
     }
 
     // synchronize PoH with a bank
-    pub fn reset(
-        &mut self,
-        tick_height: u64,
-        blockhash: Hash,
-        start_slot: Slot,
-        next_leader_slot: Option<Slot>,
-    ) {
+    pub fn reset(&mut self, blockhash: Hash, start_slot: Slot, next_leader_slot: Option<Slot>) {
         self.clear_bank();
         let mut cache = vec![];
         {
             let mut poh = self.poh.lock().unwrap();
             info!(
-                "reset poh from: {},{} to: {},{}",
-                poh.hash, self.tick_height, blockhash, tick_height,
+                "reset poh from: {},{},{} to: {},{}",
+                poh.hash, self.tick_height, self.start_slot, blockhash, start_slot
             );
             poh.reset(blockhash, self.poh_config.hashes_per_tick);
         }
 
         std::mem::swap(&mut cache, &mut self.tick_cache);
+
         self.start_slot = start_slot;
-        self.start_tick = tick_height + 1;
-        self.tick_height = tick_height;
-        self.max_last_leader_grace_ticks = self.ticks_per_slot / MAX_LAST_LEADER_GRACE_TICKS_FACTOR;
+        self.start_tick = (start_slot + 1) * self.ticks_per_slot;
+        self.tick_height = self.start_tick.saturating_sub(1);
```

Comment:
is start_tick not always at least 1? `(start_slot + 1) * self.ticks_per_slot`, `start_slot + 1 > 0` and `self.ticks_per_slot > 0`

Reply:
yep, thanks